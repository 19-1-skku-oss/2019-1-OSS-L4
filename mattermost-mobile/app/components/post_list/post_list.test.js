// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';
import {shallow} from 'enzyme';

import PostList from './post_list';
import Preferences from 'mattermost-redux/constants/preferences';

jest.useFakeTimers();

describe('PostList', () => {
    const serverURL = 'https://server-url.fake';
    const baseProps = {
        actions: {
            handleSelectChannelByName: jest.fn(),
            loadChannelsByTeamName: jest.fn(),
            refreshChannelWithRetry: jest.fn(),
            selectFocusedPostId: jest.fn(),
            setDeepLinkURL: jest.fn(),
        },
        deepLinkURL: '',
        navigator: {
            showModal: jest.fn(),
        },
        lastPostIndex: -1,
        postIds: ['post-id-1', 'post-id-2'],
        serverURL,
        siteURL: 'https://site-url.fake',
        theme: Preferences.THEMES.default,
    };

    const deepLinks = {
        permalink: serverURL + '/team-name/pl/pl-id',
        channel: serverURL + '/team-name/channels/channel-name',
    };

    const wrapper = shallow(
        <PostList {...baseProps}/>
    );

    test('should match snapshot', () => {
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('setting permalink deep link', () => {
        wrapper.setProps({deepLinkURL: deepLinks.permalink});
        expect(baseProps.actions.setDeepLinkURL).toHaveBeenCalled();
        expect(baseProps.actions.selectFocusedPostId).toHaveBeenCalled();
        expect(baseProps.navigator.showModal).toHaveBeenCalled();
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('setting channel deep link', () => {
        wrapper.setProps({deepLinkURL: deepLinks.channel});
        expect(baseProps.actions.setDeepLinkURL).toHaveBeenCalled();
        expect(baseProps.actions.handleSelectChannelByName).toHaveBeenCalled();
        expect(wrapper.getElement()).toMatchSnapshot();
    });
});
