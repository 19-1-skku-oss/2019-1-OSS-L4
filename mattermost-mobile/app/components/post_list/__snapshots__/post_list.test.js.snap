// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostList setting channel deep link 1`] = `
<FlatList
  ListFooterComponent={[Function]}
  contentContainerStyle={
    Object {
      "paddingTop": 5,
    }
  }
  data={
    Array [
      "post-id-1",
      "post-id-2",
    ]
  }
  disableVirtualization={false}
  extraData={
    Array [
      undefined,
      undefined,
      undefined,
    ]
  }
  horizontal={false}
  initialNumToRender={15}
  inverted={true}
  keyExtractor={[Function]}
  keyboardDismissMode="interactive"
  keyboardShouldPersistTaps="handled"
  maintainVisibleContentPosition={
    Object {
      "autoscrollToTopThreshold": 60,
      "minIndexForVisible": 0,
    }
  }
  maxToRenderPerBatch={16}
  numColumns={1}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onScroll={[Function]}
  onScrollToIndexFailed={[Function]}
  refreshing={false}
  removeClippedSubviews={true}
  renderItem={[Function]}
  scrollEventThrottle={60}
  updateCellsBatchingPeriod={50}
  windowSize={21}
/>
`;

exports[`PostList setting permalink deep link 1`] = `
<FlatList
  ListFooterComponent={[Function]}
  contentContainerStyle={
    Object {
      "paddingTop": 5,
    }
  }
  data={
    Array [
      "post-id-1",
      "post-id-2",
    ]
  }
  disableVirtualization={false}
  extraData={
    Array [
      undefined,
      undefined,
      undefined,
    ]
  }
  horizontal={false}
  initialNumToRender={15}
  inverted={true}
  keyExtractor={[Function]}
  keyboardDismissMode="interactive"
  keyboardShouldPersistTaps="handled"
  maintainVisibleContentPosition={
    Object {
      "autoscrollToTopThreshold": 60,
      "minIndexForVisible": 0,
    }
  }
  maxToRenderPerBatch={16}
  numColumns={1}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onScroll={[Function]}
  onScrollToIndexFailed={[Function]}
  refreshing={false}
  removeClippedSubviews={true}
  renderItem={[Function]}
  scrollEventThrottle={60}
  updateCellsBatchingPeriod={50}
  windowSize={21}
/>
`;

exports[`PostList should match snapshot 1`] = `
<FlatList
  ListFooterComponent={[Function]}
  contentContainerStyle={
    Object {
      "paddingTop": 5,
    }
  }
  data={
    Array [
      "post-id-1",
      "post-id-2",
    ]
  }
  disableVirtualization={false}
  extraData={
    Array [
      undefined,
      undefined,
      undefined,
    ]
  }
  horizontal={false}
  initialNumToRender={15}
  inverted={true}
  keyExtractor={[Function]}
  keyboardDismissMode="interactive"
  keyboardShouldPersistTaps="handled"
  maintainVisibleContentPosition={
    Object {
      "autoscrollToTopThreshold": 60,
      "minIndexForVisible": 0,
    }
  }
  maxToRenderPerBatch={16}
  numColumns={1}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onScroll={[Function]}
  onScrollToIndexFailed={[Function]}
  refreshing={false}
  removeClippedSubviews={true}
  renderItem={[Function]}
  scrollEventThrottle={60}
  updateCellsBatchingPeriod={50}
  windowSize={21}
/>
`;
